### PR TITLE
Test Automation - CEPH-83594545

### PR DESCRIPTION
**Pull request by chaitanyadommeti**
Automation for test CEPH-83594545
Added test /tests/rbd/test_rbd_group_commands_validation.py

Added test config to the suite suites/squid/rbd/tier-2_rbd_group.yaml

Success log: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-C7IEIK/

### CODE REVIEW

File: tests/rbd/test_rbd_group_commands_validation.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+                _ = pool_config.pop("data_pool")
+
+            # Create an RBD group in pool
+            group = "group_" + random_string(len=4)
+            out, err = rbd.group.create(**{"group-spec": f"{pool}/{group}"})
+            if err:
+                log.error(f"Create group {group} failed with error {err}")
+                return 1
+            else:
+                log.info(f"Successfully created pool {pool}/{group}")
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_group_commands_validation.py:68]: `It should be Group created successfully under pool {pool}/{group}`

File: tests/rbd/test_rbd_group_commands_validation.py [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+
+            # List group
+            out, err = rbd.group.list(**{"pool": pool})
+            if err:
+                log.error(f"Group list failed with error {err}")
+                return 1
+            else:
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_group_commands_validation.py:75]: this `else` can be avoided

```
if err:
    log.error(f"Group list failed with error: {err}")
    return 1

if group in out:
    log.info(f"Group {group} listed in pool {pool}")
else:
    log.error(f"Group {group} not listed in pool {pool}")
    return 1
```

File: tests/rbd/test_rbd_group_commands_validation.py [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+                log.info(f"Successfully created image {pool}/{image}")
+
+            # Add image to the group
+            out, err = rbd.group.image.add(
+                **{"group-spec": f"{pool}/{group}", "image-spec": f"{pool}/{image}"}
+            )
+            if err:
+                log.error(f"Failed to create image {pool}/{image} with error {err}")
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_group_commands_validation.py:96]: It should be `Failed to add image to the group`

File: tests/rbd/test_rbd_group_commands_validation.py [hunk 4]
```diff
@@ -0,0 +1,329 @@
...
+                return 1
+            else:
+                log.info(f"Successfully created image {pool}/{image}")
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_group_commands_validation.py:99]: please correct here as well

File: tests/rbd/test_rbd_group_commands_validation.py [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+
+            # List images in the group
+            out, err = rbd.group.image.list(**{"group-spec": f"{pool}/{group}"})
+            if err:
+                log.error(f"List images in group failed with error {err}")
+                return 1
+            else:
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_group_commands_validation.py:106]: avoid this `else` here as well.

File: tests/rbd/test_rbd_group_commands_validation.py [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+                if f"{pool}/{image}" in out:
+                    log.info(f"{pool}/{image} listed in group {pool}/{group}")
+                else:
+                    log.error(f"{pool}/{image} not listed in the group image list")
+                    return 1
+
+            # Remove the image  from the group
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_group_commands_validation.py:113]: extra space after image

File: tests/rbd/test_rbd_group_commands_validation.py [hunk 7]
```diff
@@ -0,0 +1,329 @@
...
+
+            # List images in the group
+            out, err = rbd.group.image.list(**{"group-spec": f"{pool}/{group}"})
+
+            if err:
+                log.error(f"List images in group failed with error {err}")
+                return 1
+            else:
+                if f"{pool}/{image}" not in out:
+                    log.info(f"{pool}/{image} not listed in group {pool}/{group}")
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_group_commands_validation.py:135]: group spec is used redundant can it be 
`{pool}/{image} not listed in group after successful image deletion`

File: tests/rbd/test_rbd_group_commands_validation.py [hunk 8]
```diff
@@ -0,0 +1,329 @@
...
+                else:
+                    log.error(f"{pool}/{image} listed in the {pool}/{group} image list")
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_group_commands_validation.py:137]: please correct this as well like above

File: tests/rbd/test_rbd_group_commands_validation.py [hunk 9]
```diff
@@ -0,0 +1,329 @@
...
+                return 1
+            else:
+                log.info(f"Successfully created snap {pool}/{group_new}@{snap}")
+
+            # list group snap
+            out, err = rbd.group.snap.list(**{"group-spec": f"{pool}/{group_new}"})
+            if err:
+                log.error(f"Group snap list failed with error {err}")
+                return 1
+            else:
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_group_commands_validation.py:186]: same here

File: tests/rbd/test_rbd_group_commands_validation.py [hunk 10]
```diff
@@ -0,0 +1,329 @@
...
+                if snap in out:
+                    log.info(f"Snap {snap} listed in group snap list")
+                else:
+                    log.error(f"Snap {snap} not listed in group snap list")
+                    return 1
+
+            # Delete the snap
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_group_commands_validation.py:193]: can it be `Delete the group snap`

File: tests/rbd/test_rbd_group_commands_validation.py [hunk 11]
```diff
@@ -0,0 +1,329 @@
...
+                log.info("Successfully deleted grop snap {pool}/{group_new}@{snap}")
+
+            # list group snap
+            out, err = rbd.group.snap.list(**{"group-spec": f"{pool}/{group_new}"})
+            if err:
+                log.error(f"Group snap list failed with error {err}")
+                return 1
+            else:
+                if snap not in out:
+                    log.info(f"Group snap {snap} not listed in group snap list")
```
Comment (sunilangadi2) [inline tests/rbd/test_rbd_group_commands_validation.py:210]: please add `Group snap {snap} not listed in group snap list after successful deletion`

File: tests/rbd/test_rbd_group_commands_validation.py (hunk 1 above)
Comment (chaitanyadommeti) [reply tests/rbd/test_rbd_group_commands_validation.py]: Addressed

File: tests/rbd/test_rbd_group_commands_validation.py (hunk 2 above)
Comment (chaitanyadommeti) [reply tests/rbd/test_rbd_group_commands_validation.py:77]: Addressed

File: tests/rbd/test_rbd_group_commands_validation.py (hunk 4 above)
Comment (chaitanyadommeti) [reply tests/rbd/test_rbd_group_commands_validation.py]: Addressed

File: tests/rbd/test_rbd_group_commands_validation.py (hunk 7 above)
Comment (chaitanyadommeti) [reply tests/rbd/test_rbd_group_commands_validation.py]: Addressed

File: tests/rbd/test_rbd_group_commands_validation.py (hunk 9 above)
Comment (chaitanyadommeti) [reply tests/rbd/test_rbd_group_commands_validation.py:185]: Addressed

File: tests/rbd/test_rbd_group_commands_validation.py (hunk 11 above)
Comment (chaitanyadommeti) [reply tests/rbd/test_rbd_group_commands_validation.py]: Addressed

File: tests/rbd/test_rbd_group_commands_validation.py (hunk 10 above)
Comment (chaitanyadommeti) [reply tests/rbd/test_rbd_group_commands_validation.py]: Addressed

File: tests/rbd/test_rbd_group_commands_validation.py (hunk 8 above)
Comment (chaitanyadommeti) [reply tests/rbd/test_rbd_group_commands_validation.py]: Addressed

File: tests/rbd/test_rbd_group_commands_validation.py (hunk 5 above)
Comment (chaitanyadommeti) [reply tests/rbd/test_rbd_group_commands_validation.py:107]: Addressed

**Comment by chaitanyadommeti** [comment]
Success log after implementing review comments: 
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-6BE9F8/

**Review by sunilangadi2 — APPROVED** [review]
Comments got addressed. changes LGTM.

**Review by sunilkumarn417 — APPROVED** [review]

**Pull request merged**